### ISSUE TITLE
[Feature request] Improve clarity of "Backups" and "Multi-Instance" settings

### ISSUE THREAD

**Issue by aaron-kruse** [notepad-plus-plus/notepad-plus-plus#15842]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description of the Issue

The various "Backup" and "Multi-Instance" settings offer flexibility, but with this flexibility comes some added complexities and caveats that aren't immediately obvious.

### Describe the solution you'd like.

Ideally, changes would be made to remove some of the complexities/caveats so this would no longer be an issue (ex. improve multi-instance support). However, after reviewing some of the discussions on related issues (ex. #5063), it seems like this is much easier said than done. This being the case, I would like to suggest making alternative changes to help clarify things and make these caveats more apparent to avoid running into surprises later (which could lead to things not getting backed up/restored as expected, new "bug report" issues, etc.). See below for some additional details and some specifics.

### Debug Information

```shell
<details>
  <summary>Notepad++ v8.7.1   (64-bit)</summary>
  
Notepad++ v8.7.1   (64-bit)
Build time : Oct 31 2024 - 00:48:56
Path : C:\Program Files\Notepad++\notepad++.exe
Command Line : "C:\Program Files\Notepad++\change.log"
Admin mode : OFF
Local Conf mode : OFF
Cloud Config : OFF
Periodic Backup : ON
OS Name : Windows 10 Home (64-bit)
OS Version : 22H2
OS Build : 19045.5131
Current ANSI codepage : 1252
Plugins : 
    mimeTools (3.1)
    NppConverter (4.6)
    NppExport (0.4)
    NPPJSONViewer (1.41)
</details>
```

### Feedback/suggestions
**Incorporate info from docs into Preferences UI**
- Specific areas from the Notepad++ User Manual documentation for reference:
  - [Backup](https://npp-user-manual.org/docs/preferences/#backup) ([permalink](https://github.com/notepad-plus-plus/npp-usermanual/blob/v4.5/content/docs/preferences.md#backup))
  - [Multi-Instance and Date](https://npp-user-manual.org/docs/preferences/#multi-instance-and-date) ([permalink](https://github.com/notepad-plus-plus/npp-usermanual/blob/v4.5/content/docs/preferences.md#multi-instance-and-date))
- There are several specific warnings/notes/caveats mentioned in the docs with very useful info (good), but there is no visibility of this info from the preferences UI where these settings are actually enabled/disabled (bad). The hard part here is already done, these things have already been identified and documented, but the problem is most people will probably only ever find this info when they specifically go out looking for it (after they've changed settings and after they got unexpected results, and quite possibly only after being directed to the docs when reporting a "problem"). For those like me that are willing to RTFM, it would be nice if it was referenced when/where it would be especially helpful.
- Some specific examples (emphasis mine):
  - > WARNING: The Notepad++ periodic backup feature in multi-instance setups will work only for the first Notepad++ instance; **there will be no periodic backup for subsequent instances**.
  - > Note: If you are editing two files with the same name which are in separate directories (like C:\path\a.txt and C:\other\a.txt), the Custom Backup directory **will only contain one backup file (a.txt.bak), with the contents of whichever file was saved most recently**.
  - > WARNING: If you select anything other than Default (mono-instance), changed settings in one instance **will not influence the settings in the other instance, and only the changed settings in the last instance closed will be saved to disk**.

**Update settings in the Preferences UI for clarity**
Current options for reference:
![image](https://github.com/user-attachments/assets/4d784aa1-e154-43bd-9916-15b4e2260731)

- "Remember current session for next launch"
  - This seems pretty clear at first, but the specific mention of "current session" leads to some fuzziness (ex. maybe this means the current *instance*, although if it did, then it would probably say "instance" instead of "session"... so maybe "session" means all instances currently open... then again, maybe it's a reference to the [Sessions](https://npp-user-manual.org/docs/session/#sessions) feature).
- "Enable session snapshot and periodic backup"
  - "Periodic backup" makes perfect sense. "Session snapshot" leads to some of the same fuzziness mentioned above.
  - It's also not entirely clear how this works in relation to/combination with the previous setting, and after reviewing the documentation, I see it's not quite what I would have expected.
  - Since these exist as separate options and based on the way they're worded, I would have thought the first option is solely about re-opening files on the next launch and the period backup option was solely about creating snapshot backups on a scheduled basis. After reviewing the documentation, these are kind of combo features (I'm guessing due to how features were added over time). You cannot create periodic backups without also opting in to reopening files on launch, and any unsaved files will not be reopened on next launch unless you also opt in to periodic backups.
  - None of this is really a problem for me since I want both options enabled anyways, but it just seems like some awkward dependencies between the two features given the underlying functionality that's already there.
  - Note that some of this might have been designed under the assumption a user is manually exiting, will get prompts to save, etc. but that is not always the case. For me, I always have Notepad++ up and running and rarely exit out "normally", it's a Windows update that does an auto-reboot overnight that triggers the exit (so I won't see any prompts that might otherwise normally happen). 
- "Remember inaccessible files from past session"
  - It seems very backwards to have this disabled by default. This can be a very helpful feature that helps you identify potential problems sooner so you can take action if necessary. By being disabled by default, people are less likely to realize this feature even exists and only find it after having ran into problems (especially if they enabled "remember current session" before this feature was added).

Lastly, I don't mean for any of this to come off as negative or me complaining. I appreciate all the hard work and I wouldn't have taken the time to write all this up if it were only to complain about things. I just wanted to share some thoughts/ideas that might help make this great app a little better :)
